### ISSUE TITLE
Issues with scoped containers and Tokens

### ISSUE THREAD

**Issue by mscharley** [typestack/typedi#52]
I have code similar to the following:

```
export interface TestService {
  sayHello(): void;
}
export const TestService = new Token<TestService>("TestService");

@Service({ id: TestService, factory: () => new TestServiceImpl() })
class TestServiceImpl {
  public sayHello() {
    console.log("Hello world!");
  }
}
```

Then I can't run the following:

```
Container.get(TestService).sayHello();
// Works.
Container.of({ foo: "bar" }).get(TestService).sayHello();
// ServiceNotFoundError: Service "TestService" was not found, looks like it was not registered in the container. Register it by calling Container.set before using service.
```

An explicit set in the scoped container fixes the issue but also duplicates the service definition.

**Comment by pleerock** [comment]
what typedi version are you using?

**Comment by mscharley** [comment]
0.6.1

**Comment by pleerock** [comment]
try latest, 0.7.0

**Comment by mscharley** [comment]
That seems to have fixed it - but neither 0.6.1 nor 0.7.0 was in Github releases where many other versions are tagged so I didn't realise they were out (NPM grabbed 0.6.1 a while ago automatically). Cheers though!
